packages/contractkit/src/explorer/base.ts

```diff
@@ -23,16 +31,43 @@ export const getContractDetailsFromContract = async (
   }
 }
 
-export async function obtainKitContractDetails(kit: ContractKit): Promise<ContractDetails[]> {
-  return concurrentMap(5, RegisteredContracts, (celoContract) =>
+const getContractMappingFromDetails = (
+  cd: ContractDetails,
+  abiFilterType: AbiType
+): ContractMapping => ({
+  details: cd,
+  abiMapping: mapFromPairs(
+    cd.jsonInterface
+      .concat(PROXY_ABI)
+      .filter((ad) => ad.type === abiFilterType)
+      .map((ad) => [ad.signature, ad])
+  ),
+})
+
+const obtainKitContractDetails = (kit: ContractKit): Promise<ContractDetails[]> =>
+  concurrentMap(5, RegisteredContracts, (celoContract) =>
     getContractDetailsFromContract(kit, celoContract)
   )
-}
 
-export function mapFromPairs<A, B>(pairs: Array<[A, B]>): Map<A, B> {
-  const map = new Map<A, B>()
-  pairs.forEach(([k, v]) => {
-    map.set(k, v)
-  })
-  return map
+const getAddressMappingFromDetails = (contractDetails: ContractDetails[], abiType: AbiType) =>
+  mapFromPairs(
+    contractDetails.map((cd) => [cd.address, getContractMappingFromDetails(cd, abiType)])
+  )
+
+export class BaseExplorer {
```

Comment:
I like this. Do you foresee a use for `getContractDetailsFromContract`, `obtainKitContractDetails` or the other methods to be used outside of classes? It's nice to have the option to only import a method and use it vs extending a class, but if it isn't used outside of the context, then this LGTM.

Reply:
exported for any future use cases